ann/ann.py

```diff
@@ -0,0 +1,113 @@
+# ============================================================================
+# Neural Network functions using TensorFlow and Keras
+# Author : Valérie Bibeau, Polytechnique Montréal, H4ck4th0n 2023
+# ============================================================================
+
+# ---------------------------------------------------------------------------
+# Imports
+# Database
+import pandas as pd
+# Arrays for data
+import numpy as np
+# To normalize data
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.model_selection import train_test_split
+# NN
+np.random.seed(1)               # for reproducibility
+import tensorflow
+tensorflow.random.set_seed(2)   # for reproducibility
+from tensorflow import keras
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import Dense
+# ----------------------------------------------------------------------------
+
+def read_data(file_path):
+    """Read the database
+
+    Args:
+        file_path (string): Name of the path where the database is
+
+    Returns:
+        X (array): Features tensor
+        Y (array): Outputs tensor
+    """
+
+    data = pd.read_csv(file_path, index_col=0)
+    data = data.to_numpy()
+    X = data[:,:-1]
+    Y = data[:,-1].reshape(-1,1)
+
+    return X, Y
+
+def initial_setup(X, Y, test_size, random_state):
+    """Set up the training and testing set
+
+    Args:
+        X (array): Features tensor
+        Y (array): Outputs tensor
+        random_state (int): Random number to split the training and testing set
+
+    Returns:
+        X and Y: Features and target values of the training and testing set
+    """
+    # Normalizing features
+    scaler_X = MinMaxScaler()
+    scaler_Y = MinMaxScaler()
+    scaler_X.fit(X)
+    scaler_Y.fit(Y)
+    Xscale = scaler_X.transform(X)
+    Yscale = scaler_Y.transform(Y)
+    # Split the data into training and testing
+    X_train, X_test, Y_train, Y_test = train_test_split(Xscale, Yscale, 
+                                                        test_size=test_size,
+                                                        random_state=random_state)
+    
+    return X_train, X_test, Y_train, Y_test, scaler_X, scaler_Y
+
+def fit_model(X_train, Y_train,
+              units, layers, activation, lr,
+              epochs, batch_size, val_frac, l2,
+              verbose):
+    """Neural Network architecture/model
+
+    Args:
+        X_train and Y_train: Features and target values of the training set
+        units (int): Number of units of the first hidden layer
```

Comment:
```suggestion
        units (int): Number of units in the hidden layer
```
You use units for all layers no?